File: webOfferPro/src/app/models/offer_pro_model/RecruitmentOfferDto.ts
import { Gender } from "../user_model/Gender";
import { ApplyOnOfferDto } from "./ApplyOnOfferDto";
import { CompanyDto } from "./CompanyDto";
import { ContractType } from "./ContractType";
import { FormDto } from "./FormDto";
import { StatusOffer } from "./StatusOffer"; 
import { StudyLevel } from "./StudyLevel";
import { TypeOffer } from "./TypeOffer";

export class RecruitmentOfferDto {
    id! : number ;
    typeOffer! : TypeOffer ;
    title! : string ;
    description! : string ;
    requirements! : string ;
    contractType! : ContractType ; 
    endDateTime :  Date = new Date();
    startDateTime : Date = new Date();
    statusOffer! : StatusOffer;
    vacantJobs! : number;
    gender! :  Gender; 
    experienceFrom! : number;
    experienceTo! : number;
    langue! : string ;
    studyLevel! : StudyLevel;
    campanyDto : CompanyDto=  new CompanyDto();;
    applyOnOffersDto : ApplyOnOfferDto[]  = new Array<ApplyOnOfferDto>();
    forms : FormDto[] = new Array<FormDto>();  ;
    constructor() {}
} 

File: plateforme/offer-pro-service/src/main/java/tn/esprit/Dtos/RecruitmentOfferDto.java
package tn.esprit.Dtos;

import lombok.*;
import lombok.experimental.FieldDefaults;
import tn.esprit.Entitys.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;



@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@FieldDefaults(level = AccessLevel.PRIVATE)
public class RecruitmentOfferDto {
    long id;
    TypeOffer typeOffer;
    String title;
    int vacantJobs;
    String description;
    String requirements;
    ContractType contractType;
    Gender gender;
    String langue;
    int experienceFrom;
    int experienceTo;
    LocalDateTime startDateTime;
    LocalDateTime endDateTime;
    StatusOffer statusOffer;
    Studylevel studyLevel;
    CompanyDto campanyDto;
    List<ApplyOnOfferDto>  applyOnOffersDto = new ArrayList<ApplyOnOfferDto>();
    List<Form>  forms = new ArrayList<Form>();
}
